t/lib/OpenQA/SeleniumTest.pm

```diff
@@ -352,9 +352,7 @@ sub get_mojoport () { $mojoport }
 # uncoverable subroutine
 # uncoverable statement
 sub driver_missing () {
-    diag 'Install Selenium::Remote::Driver and Selenium::Chrome to run these tests';    # uncoverable statement
-    done_testing;    # uncoverable statement
-    exit;    # uncoverable statement
+    plan skip_all => 'Install Selenium::Remote::Driver and Selenium::Chrome to run these tests'; # uncoverable statement
```

Comment:
```suggestion
    my $msg = 'Install Selenium::Remote::Driver and Selenium::Chrome to run these tests';
    plan skip_all => $msg unless $exit; # uncoverable statement
    diag $msg;    # uncoverable statement
    done_testing;    # uncoverable statement
    exit;    # uncoverable statement
```

Reply:
I don't really love that idea. `skip_all` already handles printing the message and skipping; it's a one-stop shop. Doing `diag $msg` after `plan skip_all => $msg` would duplicate the message. Calling `done_testing` after `skip_all` is of questionable validity (it might even cause an error, I haven't checked); it's certainly redundant.

We could just make it a big if/else:

    sub driver_missing {
        my $exit = shift;
        my $msg = 'Install Selenium::Remote::Driver and Selenium::Chrome to run these tests';
        if ($exit) {
            diag $msg;
            done_testing;
            exit;
        }
        else {
            plan skip_all => $msg;
        }
    }

but since we only need the differing behaviour in one place, is there actually any point in factoring it out?

Reply:
(this would, of course, be a lot simpler if `Test::More` had a `skip_remaining`, but AFAICT it doesn't. sigh. why is this test framework so weird about this stuff?)

Reply:
Huh? You are right now duplicating the message, that was not my idea :-D I only suggested to handle it once in driver_missing so that there is no need for that

Reply:
> We could just make it a big if/else:

Ah. Yes. I guess there's a typo in my snippet. I just meant to handle both in one place, no change in the logic.
If you're spelling out the same $msg twice it's likely to be changed inconsistently or copied if needed for another case. It could be as simple as a second unit test to avoid making this one too big which we sometimes do.

Reply:
oh, I see. then yes, if you think not duplicating the message is important, we can do it that way. there are some pointers in my upstream issue to a way we could have driver_missing discover which path it needs to use - I'm going to look at that when I get a minute.

Reply:
new version of the PR avoids these concerns, I think.

Reply:
This looks way slicker. Thank you!